### ISSUE TITLE
fresh init has broken test

### ISSUE THREAD

**Issue by nazreen** [project-serum/anchor#1791]
to repro: init a new anchor project and run `anchor test`

you'll encounter: `Error: failed to send transaction: Transaction simulation failed: Attempt to load a program that does not exist`

context:
- solana-cli 1.10.8 (src:623ac656; feat:1122441720)
- M1 Macbook Pro, mac OS 12.1
- anchor-cli 0.24.2

**Comment by armaniferrante** [comment]
Would be great to write a test for this once fixed.

**Comment by paul-schaaf** [comment]
Ive added a test for this a while ago (`test-anchor-init` in our ci) and I'm not also not able to reproduce it.

My guess is you're using an older anchor version and already have a local validator running when running `anchor test`?

**Comment by nazreen** [comment]
@paul-schaaf thank you for your reply.

I've amended the original issue message to mention the versions of solana-cli and anchor-cli. 
My anchor-cli is on 0.24.2

So when I reported the issue, I was running with the `--skip-local-validator` flag which I assumed was meant to avoid the 'clash' - is this not the case?

Just now I did run the usual `anchor test` while my local validator was off, and there was no error indeed. Thank you.

But back to the flag, is that flag not mean to be a way to make tests still work despite a local validator already running?
